docs/state-hydration-snapshots.md

```diff
@@ -0,0 +1,52 @@
+<!-- @todo this is all very technical, might use rewording -->
+
+<!--notes-->
+
+<!--hydration
+- Broker::commit method
+- can see what happens on the back end
+- runs handle on the events, stores all the states on snapshots store
+
+- read state::load method (what happens at the beginning)
+
+- can maybe show the json blob of a dehydrated state?
+
+
+- serialization explanation
+    - Verbs serializes and deserializes things to strings to be stored
+    - We have default serializers in the verbs.config
+    - You can add another type of object like this:
+        - SerializeByVerbs interface
+        - And a Trait called something like "SerializesToPropertyNamesAndValues"
+            - I'll take all this and serialize to JSON
+
+    - Like synths but for verbs
+        - like a DTO on a state would need a new serializer
+        - Or the LineItemCollection from invoices-->
+
+## Hydration
+
+In Verbs, we use state snapshots to conveniently hydrate states in memory.
+
+When you `load()` a State (or retrieve a `singleton()` state)
+- If a snapshot exists, the state is hydrated by loading and deserializing the snapshot data.
+    - If not, it initiates event replay to rebuild the state based on currently stored events.
```

Comment:
It's not really replaying events… it's only re-applying them to the state. I worry that if we say replay people will think that `handle` is called.